### DISCUSSION TITLE
Not able to deploy on vercel. any help?

### DISCUSSION THREAD

**Discussion by clickableman1** [tjtanjin/react-chatbotify#77] [Q&A]
It shows the code 

![Screenshot 2024-07-05 at 6 23 39 AM](https://github.com/tjtanjin/react-chatbotify/assets/101219069/c16ddc20-d5f3-4862-a8f0-53912e574109)

**Comment by tjtanjin** [comment]
Hey @clickableman1, what's the error message you're receiving?

**Comment by clickableman1** [comment]
![image](https://github.com/tjtanjin/react-chatbotify/assets/101219069/de5a6135-7cc1-4985-84a2-bb6829941209)

No errors on the build. when I open it on the browser, it shows the code.

**Reply by clickableman1** [reply]
@tjtanjin

**Comment by tjtanjin** [comment]
Hmmm I can't really tell what you're trying to do and I've not used vercel for deployment before... Is your project somewhere on GitHub where I can take a look? Or maybe you can share what you're trying to do?

**Reply by clickableman1** [reply]
I have cloned the repo and deployed it on Vercel. It's showing the code instead of chatbot. 
Could you share some hosting providers that I could use? I did use Digitalocean too. 

The repo is private and I am using main branch.

**Reply by tjtanjin** [reply]
> I have cloned the repo and deployed it on Vercel. It's showing the code instead of chatbot. Could you share some hosting providers that I could use? I did use Digitalocean too.
> 
> The repo is private and I am using main branch.

If I'm understanding your correctly, you cloned the React ChatBotify repository and tried deploying it directly? If that's the case it wouldn't work because it is built in vite library mode. You'll have to create your own project then import the chatbot library for deployment.

**Reply by clickableman1** [reply]
I understand, do you have any examples for that? I want to use OpenAI with that.  I would be very grateful to you. 
Thanks

**Reply by tjtanjin** [reply]
> I understand, do you have any examples for that? I want to use OpenAI with that. I would be very grateful to you. Thanks

The documentation has live examples for your use case. The following 2 examples deal with OpenAI/Gemini:
- https://react-chatbotify.tjtanjin.com/docs/examples/real_time_stream
- https://react-chatbotify.tjtanjin.com/docs/examples/llm_conversation
